### PR TITLE
Use size.x element for hud text as the font size.

### PR DESCRIPTION
**Pull request by basicer**

### CODE REVIEW

**Comment by basicer** [comment]
I had also considered adding a new font size property as a short, as was done in a previous pull request, not sure which is preferable.

**Comment by SmallJoker** [comment]
The player's default font size should be respected, as the by-server specified size may be too small or too big.
I suggest using `scale`:
```C++
u32 fontsize = g_fontengine->getDefaultFontSize();
if (e->scale.X != 0)
	fontsize = std::round(e->scale.X * fontsize);
irr::gui::IGUIFont *font = g_fontengine->getFont(fontsize, FM_Unspecified);
```

File: src/hud.cpp [hunk 1]
```diff
@@ -318,6 +317,9 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 										 (e->number >> 0)  & 0xFF);
 				core::rect<s32> size(0, 0, e->scale.X, text_height * e->scale.Y);
 				std::wstring text = unescape_translate(utf8_to_wide(e->text));
+				s32 fontsize = FONT_SIZE_UNSPECIFIED;
+				if (e->size != v2s32()) fontsize = e->size.X;
```
Comment (SmallJoker) [inline src/hud.cpp]: `fontsize = ` belongs on a new line.

File: src/hud.cpp [hunk 2]
```diff
@@ -318,6 +317,9 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
...
+				irr::gui::IGUIFont* font = g_fontengine->getFont(fontsize, FM_Unspecified);
```
Comment (SmallJoker) [inline src/hud.cpp]: Misplaced pointer: `irr::gui::IGUIFont *font`

File: src/hud.cpp [hunk 3]
```diff
@@ -355,6 +357,7 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 										 (e->number >> 0)  & 0xFF);
 				core::rect<s32> size(0, 0, 200, 2 * text_height);
 				std::wstring text = unescape_translate(utf8_to_wide(e->name));
+				irr::gui::IGUIFont* font = g_fontengine->getFont();
```
Comment (SmallJoker) [inline src/hud.cpp]: `irr::gui::IGUIFont *font`

**Comment by basicer** [comment]
It uses their default font size if unspecified.

It could multiply the requested font size by the ratio between the minetest default font size and the requested font size, but that seems a bit magical.

**Comment by xunto** [comment]
I think that instead of letting mod change font size, it would be better to allow mods to read user-defined font size to adjust ui.

**Comment by rubenwardy** [comment]
Please multiply the teztsize, eg make X a decimal where 100 is the current value and 200 is twice the size

**Comment by paramat** [comment]
^ @basicer

**Comment by ClobberXD** [comment]
> Please multiply the teztsize

@rubenwardy Do you mean that `size.x` should be multiplied with `FONT_SIZE_UNSPECIFIED`?

**Comment by paramat** [comment]
https://github.com/minetest/minetest/pull/6560#issuecomment-353594207

No response from author so closing.

**Pull request closed without merging**